src/NuGet.Services.Validation.Orchestrator/ContentScan/ContentScanValidator.cs

```diff
@@ -0,0 +1,142 @@
+            }
+
+            _configuration = configurationAccessor.Value;
+        }
+
+        public async Task<IValidationResponse> StartAsync(IValidationRequest request)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            var validatorStatus = await GetValidatorStatusAsync(request);
+
+            if (validatorStatus.State != ValidationStatus.NotStarted)
+            {
+                _logger.LogWarning(
+                    "Scan and Sign validation with validation Id {ValidationStepId} has already started.",
+                    request.ValidationStepId);
+
+                return GetValidationResponse(validatorStatus);
+            }
+
+            if (ShouldSkipScan(request))
+            {
+                return ValidationResponse.Succeeded;
+            }
+
+            await _contentScanEnqueuer.EnqueueContentScanAsync(request.ValidationStepId, request.InputUrl.OriginalString);
+
+            var result = await _validatorStateService.TryAddValidatorStatusAsync(request, validatorStatus, ValidationStatus.Incomplete);
+
+            var response = GetValidationResponse(result);
+            return response;
+        }
+
+        public async Task CleanUpAsync(IValidationRequest request)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+            //TODO: need to check if cvs blobs should be clear here or by CVSJob(prefer)
+        }
+
+        public async Task<IValidationResponse> GetResponseAsync(IValidationRequest request)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            var result = await GetValidatorStatusAsync(request);
+
+            var response = GetValidationResponse(result);
+            return response;
+        }
+
+        private async Task<ValidatorStatus> GetValidatorStatusAsync(IValidationRequest request)
+        {
+            var validatorStatus = await _validatorStateService.GetStatusAsync(request);
+            return validatorStatus;
+        }
+
+        private bool ShouldSkipScan(IValidationRequest request)
+        {
+            return false;
+        }
+
+        private IValidationResponse GetValidationResponse(ValidatorStatus status)
+        {
+            if (status.State == ValidationStatus.Failed)
+            {
+                var packageValidationresult = new PackageValidationResult();
+                packageValidationresult.Type = "ContentViolation";
+                packageValidationresult.Data = "{}";
+                List<PackageValidationResult> resultList = new List<PackageValidationResult>() { packageValidationresult };
+                var response = new ValidationResponse(status.State, resultList);
+                return response;
+
```

Comment:
Considering simplifying to:

```suggestion
                var results = new List<PackageValidationResult>
                {
                    new PackageValidationResult
                    {
                        Type = "ContentViolation",
                        Data = "{}",
                    }
                };

                return new ValidationResponse(status.State, results);
```

I don't feel strongly about this. If you prefer your current code, please use `var` and capitalize the "r" in `packageValidationResult`